Src/FluentAssertions/Common/ExpressionExtensions.cs

```diff
@@ -87,15 +87,15 @@ internal static class ExpressionExtensions
                 case ExpressionType.Call:
                     var methodCallExpression = (MethodCallExpression)node;
 
-                    if (methodCallExpression.Method.Name != "get_Item" || methodCallExpression.Arguments.Count != 1 ||
-                        methodCallExpression.Arguments[0] is not ConstantExpression)
+#pragma warning disable SA1010 // https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/3507
+                    if (methodCallExpression is not { Method.Name: "get_Item", Arguments: [ConstantExpression argumentExpression] })
```

Comment:
Difference:
Null checks on `methodCallExpression`, `methodCallExpression.Method` and `methodCallExpression.Arguments`

[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4B0AMgJYB2AjpQKIIAOsAzl7RPVwDchQhSSkIwAFYwwAF1IA1AIYAbWgBNlcmAFkYAW2AwoABW0ALAHIwA7gB4AKgBFZq5VAYBzRwE92MAA0pI6mUBABUHK+fgEAfAAUbJwwPHz09hgArE6uYO6e9D7+QSFhESbRsTBxcaQwHNy8/ACUhADehKTdpMlN6aT0EBowpAC89Y2pzfTCBD2kXLa0cmAWpAlDI5RWwzDVbfM9nUcL3WDKXKN90+nVlADCaqogXWfdAG4epAYwchbDJ6qVQ3NL8cYbfT/QHPUEzFpbGBzN7vWgAMw2v2hGiBIKmYPopFoXEGEAU7VIUIBGh2yl+IFIACIvH8APoASR0BkZwQAglAvABXX70ORcBkAbQe/C4cmUorhAw8QpFckV/AAuqQAL6Hd7dE76s7/cK2QZ2Uj8lUwBUIMAwdhydIJFpzI3alFnT0LDAAdlIyuFNrV+JmlBUqkFSM9HtOpE9fsGguBc1johw4kkMnkSjUmm0ekMxjMlgA8qoNLk3B5vNVgqFwpEqiVEuqMtkq/ka0U62VG5UYi26g0UgTDoaem3SSMISP+vw3T0lis1htETs9gdPRP3hcrr1Q3cSo9nq84wsvlAfn9qbipxMElSYcC2wi9sjzz10Zib8+8aOwyfGkrDpUYAEIJmZNlOUMRlSDQNBr2xO9D34SgrSDUUuEeCBBVFUgINIHB4LQb0jXI7osVvWFUPodCBUwsUJWILViVJBRpQEOVbQA9I9X1HdyJNCAzXoC0MNVNh7UdZ1XTI7pYwo7p5NIS9SEgLj5RDXjwQfTjZS018qL/Nt6OtLDmI1D8jUTDSDJ4+c6IjKNFwUz0E39ehk1UVz4wIbUgA)

![image](https://user-images.githubusercontent.com/919634/229341317-3cec8990-5429-4b40-b6f4-e6e0ca6c1c4b.png)